### PR TITLE
Changes SR to not be an instant and unavoidable revival.

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do
Strange reagent will now prompt the dead person to reenter their body instead of instantly reviving them. It also makes them jitter a bit to have a visual effect while the process is taking place.

## Why It's Good For The Game
Being forced back into your body sucks because the only other option is DNR.

## Images of changes

Feeding a skrell and reviving them

https://user-images.githubusercontent.com/91113370/187049875-693d845a-0871-4631-acaa-da9c862cd58a.mp4

## Testing

Revived a skrell with the ghost inside the body
Revived a skrell with the ghost outside the body, but moved inside the body before revival period ended.
Failed to revive a skrell with the ghost not reentering the body.

## Changelog
:cl:
tweak: Ghosts will now be prompted to reenter their body when being revived with SR, instead of being forced back to life.
tweak: It takes about 5 seconds for Strange Reagent to take effect.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/medicine.dm [hunk 1]
```diff
@@ -767,38 +768,55 @@
...
-						H.update_body()
-
-					M.grab_ghost()
-					M.update_revive()
-					add_attack_logs(M, M, "Revived with strange reagent") //Yes, the logs say you revived yourself.
+				M.visible_message("<span class='warning'>[M] shakes violently!</span>")
+				if(ghost && ghost.can_reenter_corpse && ghost.client)
+					to_chat(ghost, "<span class='ghostalert'>Your body is being revived with Strange Reagent. Return to your body if you want to be revived!</span>")
+					window_flash(ghost.client)
+					ghost << sound('sound/effects/genetics.ogg')
```
Comment (Edan52) [inline code/modules/reagents/chemistry/reagents/medicine.dm:782]:
```suggestion
					SEND_SOUND(ghost, sound('sound/effects/genetics.ogg'))
```

**Comment by Contrabang** [comment]
Having no client and being revived with SR works like it used to now

**Comment by Denthamos** [comment]
Does this have any impact in regards to SR on non humanoid mobs?

**Comment by Contrabang** [comment]
Hmm, yeah I thought I hadn't affected simple mob revival code but I changed the way the ghost is grabbed for all of them. I will need to make more changes to that, maybe include a similar revival thing for them. I may have also messed with non-human carbons (like xenomorphs). I will apply fixes for that shortly.

**Comment by Contrabang** [comment]
I did some testing, SRing simplemobs works like it it did before, with no cooldown and only on splashing. On carbon/aliens it seems to work fine too, gibbing and reviving works on the xenos.

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**